### PR TITLE
Add exp benchmarks

### PR DESCRIPTION
**Pull request by musm**
May need latest BenchmarkTools to get more meaningful timings on nanosoldier?

### CODE REVIEW

File: src/scalar/ScalarBenchmarks.jl [hunk 1]
```diff
@@ -145,4 +145,30 @@ for b in values(g)
...
+g["exp","overflow",                        "Float 64"] = @benchmarkable exp(900.0)
+g["exp","underflow",                       "Float 64"] = @benchmarkable exp(-900.0)
+
+g["exp","normal path, k = 2",              "Float 32"] = @benchmarkable exp(1f5)
+g["exp","fast path, k = 1",                "Float 32"] = @benchmarkable exp(0f5)
+g["exp","no agument reduction, k = 9",     "Float 32"] = @benchmarkable exp(0f1)
+g["exp","small argument path",             "Float 32"] = @benchmarkable exp(2f0^-15)
+g["exp","normal path -> small, k = -1045", "Float 32"] = @benchmarkable exp(-724f0)
+g["exp","overflow",                        "Float 32"] = @benchmarkable exp(150f0)
+g["exp","underflow",                       "Float 32"] = @benchmarkable exp(-150f0)
```
Comment (jrevels) [inline src/scalar/ScalarBenchmarks.jl]: Can we keep these in the same group as the other scalar exponentiation benchmarks? Maybe just change the current `"floatexp"` group to `"exp"`?

File: src/scalar/ScalarBenchmarks.jl [hunk 2]
```diff
@@ -145,4 +145,30 @@ for b in values(g)
     b.params.time_tolerance = 0.40
 end
 
+#######
+# exp #
+#######
+
+g = addgroup!(SUITE, "exp")
+
+g["exp","normal path, k = 2",              "Float 64"] = @benchmarkable exp(1.5)
```
Comment (jrevels) [inline src/scalar/ScalarBenchmarks.jl]: You can leave out the additional `"exp"` here. It's redundant since `"exp"` is already the group ID.

**Comment by musm** [comment]
k fixed thanks

**Comment by jrevels** [comment]
Sorry for the delay - these benchmarks are now on Nanosoldier.

**Pull request merged**